### DISCUSSION TITLE
A new strategy "Context"

### DISCUSSION THREAD

**Discussion by SDGLBL** [olimorris/codecompanion.nvim#112] [Ideas]
[zed.ai](https://zed.dev/blog/zed-ai )

I tried out zedai, and the way it combines chat and inline into a single context is very useful. This method allows users to interact with the model in the chat window to finalize modification plans, and then apply these modifications one by one using the inline method in the buffer.

**Comment by olimorris** [comment]
Thanks for sharing. That is very interesting indeed. I'm currently refactoring the way messages are stored in the chat buffer which might open up functionality similar to this.

**Reply by SDGLBL** [reply]
@olimorris  I've just submitted a demo-style pull request, which I hope will be helpful in your implementation of this feature. Additionally, the workflow in Zed has left a strong impression on me, and I think you might want to take a look at that as well. 

https://github.com/olimorris/codecompanion.nvim/pull/114

**Reply by olimorris** [reply]
Thanks for sharing. I'll be taking a closer look at this in the next week. Really appreciated.

**Reply by olimorris** [reply]
@SDGLBL I've added this as per [v1.3.0](https://github.com/olimorris/codecompanion.nvim/discussions/130)
